frontend/src/components/event-handler.tsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import {
+  useFetcher,
+  useLoaderData,
+  useRouteLoaderData,
+} from "@remix-run/react";
+import { useDispatch, useSelector } from "react-redux";
+import toast from "react-hot-toast";
+
+import {
+  useWsClient,
+  WsClientProviderStatus,
+} from "#/context/ws-client-provider";
+import { ErrorObservation } from "#/types/core/observations";
+import { addErrorMessage, addUserMessage } from "#/state/chatSlice";
+import { handleAssistantMessage } from "#/services/actions";
+import {
+  getCloneRepoCommand,
+  getGitHubTokenCommand,
+} from "#/services/terminalService";
+import {
+  clearFiles,
+  clearSelectedRepository,
+  setImportedProjectZip,
+} from "#/state/initial-query-slice";
+import store, { RootState } from "#/store";
+import { createChatMessage } from "#/services/chatService";
+import { Settings } from "#/services/settings";
+import { clientLoader as rootClientLoader } from "#/routes/_oh";
+import { isGitHubErrorReponse } from "#/api/github";
+import OpenHands from "#/api/open-hands";
+import { base64ToBlob } from "#/utils/base64-to-blob";
+import { setCurrentAgentState } from "#/state/agentSlice";
+import AgentState from "#/types/AgentState";
+
+interface ServerError {
+  error: boolean | string;
+  message: string;
+  [key: string]: unknown;
+}
+
+const isServerError = (data: object): data is ServerError => "error" in data;
+
+const isErrorObservation = (data: object): data is ErrorObservation =>
+  "observation" in data && data.observation === "error";
+
+interface EventHandlerProps {
+  children?: React.ReactNode | undefined;
+}
+
+interface LoaderData {
+  settings: Settings;
+  token: string;
+  ghToken: string;
+  repo: string;
+  q: string;
+  lastCommit: string;
+}
+
+export function EventHandler({ children }: EventHandlerProps) {
+  const { events, status, send } = useWsClient();
+  const statusRef = React.useRef<WsClientProviderStatus | null>(null);
+  const runtimeActive = status === WsClientProviderStatus.ACTIVE;
+  const fetcher = useFetcher();
+  const dispatch = useDispatch();
+  const { files, importedProjectZip } = useSelector(
+    (state: RootState) => state.initalQuery,
+  );
+  const { ghToken, repo }: LoaderData = useLoaderData();
```

Comment:
After adding `import { clientLoader as appClientLoader } from "#/routes/_oh.app";`:
```suggestion
  const { ghToken, repo } = useLoaderData<typeof appClientLoader>();
```

You can then remove the `LoaderData` interface in preference of the source type

Reply:
Done!